cmd/access/node_builder/staked_access_node_builder.go

```diff
@@ -210,13 +212,19 @@ func (builder *StakedAccessNodeBuilder) initLibP2PFactory(ctx context.Context,
 	}
 
 	return func() (*p2p.Node, error) {
+		psOpts := p2p.DefaultPubsubOptions(p2p.DefaultMaxPubSubMsgSize)
+		psOpts = append(psOpts, func(_ context.Context, h host.Host) (pubsub.Option, error) {
+			return pubsub.WithSubscriptionFilter(p2p.NewSubscriptionFilter(
```

Comment:
I feel a one-liner comment on how the magic happens would help: here, the SANs have the libp2p peers in their `IdentityProvider` and would therefore let unstaked subscriptions through.

Reply:
That's not exactly how it works actually. In fact, the subscription behaves the exact same way on all nodes.

The subscription filter logic always allows unstaked nodes to subscript to unstaked channels.

Reply:
I think we're saying the same thing, and just I fumbled my wording, thinking, or both.